### ISSUE TITLE
Source fails to build on OSX and Ubuntu Linux

### ISSUE THREAD

**Issue by daden** [forcedotcom/aura#60]
My issue looks similar to: https://github.com/forcedotcom/aura/issues/38

I'm following the directions in: https://github.com/forcedotcom/aura/blob/master/aura_oss.pdf

I got the helloWorld app to work and added the new hello.cmp component and then moved on to building the source. I cloned from github, went into the directory and ran maven install but get a failure on aura-impl. I started in OSX El Capitan, then set up an Ubuntu 16.04 virtual machine to see if the problem was platform-related. I get virtually the same error on both: 

[INFO] 
[INFO] --- frontend-maven-plugin:0.0.28:npm (run-xunit) @ aura-impl ---
[INFO] Found proxies: []
[INFO] Running 'npm run --color=false test -- /dependency:./aura-util/src/test/tools/xUnit/dependencies /output:text /strict:true /verbose:false /home/daden/lightening/aura/aura/aura-impl/src/test/javascript' in /home/daden/lightening/aura/aura/aura-impl/..
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO] 
[INFO] Aura Framework ..................................... SUCCESS [  6.082 s]
[INFO] aura-interfaces .................................... SUCCESS [  4.232 s]
[INFO] aura-spring ........................................ SUCCESS [  1.734 s]
[INFO] aura-util .......................................... SUCCESS [ 25.552 s]
[INFO] aura ............................................... SUCCESS [ 14.871 s]
[INFO] aura-impl-expression ............................... SUCCESS [ 13.774 s]
[INFO] aura-impl .......................................... FAILURE [01:54 min]
[INFO] aura-tools ......................................... SKIPPED
[INFO] aura-components .................................... SKIPPED
[INFO] aura-resources ..................................... SKIPPED
[INFO] aura-integration-test .............................. SKIPPED
[INFO] aura-jetty ......................................... SKIPPED
[INFO] archetype for Aura-based "hello, world" ............ SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 03:01 min
[INFO] Finished at: 2016-09-12T15:31:30-04:00
[INFO] Final Memory: 104M/287M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal com.github.eirslett:frontend-maven-plugin:0.0.28:npm (run-xunit) on project aura-impl: Failed to run task: 'npm run --color=false test -- /dependency:./aura-util/src/test/tools/xUnit/dependencies /output:text /strict:true /verbose:false /home/daden/lightening/aura/aura/aura-impl/src/test/javascript' failed. java.io.IOException: Cannot run program "/home/daden/lightening/aura/aura/aura-impl/../node/node" (in directory "/home/daden/lightening/aura/aura/aura-impl/.."): error=2, No such file or directory -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
[ERROR] 
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <goals> -rf :aura-impl

In the aura-impl file, I comment out the failing test (below) and was able to get to successful completion. 

<configuration>
    <arguments>run --color=false test -- /dependency:./aura-util/src/test/tools/xUnit/dependencies /output:text /strict:${xunit.js.isStrict} /verbose:false ${project.basedir}/src/test/javascript</arguments>
</configuration>

**Comment by rahul-chaudhary** [comment]
It is trying to prepare node for js test running and it is failing in process because it is not able to find node available at this. please go to pom.xml for failing module aura-impl in this case and look for node inside 
<build>
<plugins>
search for frontend-maven-plugin ... change the path of node to your local also change version for node and npm as per your local. 

</plugins>
</build>

you also might need to main aura project pom.xml (parent). try and let me know. in worst case we can always set skipTestCases to false to build it successfully.

**Comment by wolfch** [comment]
Hello, I am having the same issue.  Firstly, there is no mention of `node` or `npm` requirement in the [aura_oss.pdf](https://github.com/forcedotcom/aura/blob/master/aura_oss.pdf) document in the section, "Quick Start".  Secondly, if I copy & paste the failed command:
```
$ cd aura-impl
$ npm run --color=false test -- /dependency:./aura-util/src/test/tools/xUnit/dependencies /output:text /strict:true /verbose:false /Users/wolfch/src/aura/aura-impl/src/test/javascript
```
...and run from the command line, then that works. (because I happen to have node installed and on my PATH).  Obviously, it would be preferable to not skip unit tests.

**Comment by wolfch** [comment]
Hi, I was able to get past this issue by running the following Maven goal first:
```
$ mvn pre-clean
```
This will apparently install a version of `node`, `npm` under a directory `${project.basedir}/node` and `grunt` module under a directory `${project.basedir}/node_modules`.  In other words, it will use it's own downloaded version of `node` and `npm` regardless of what you have installed, if any.

I did run `mvn clean` per the instructions, but I suppose this didn't implicitly call `mvn pre-clean`.
